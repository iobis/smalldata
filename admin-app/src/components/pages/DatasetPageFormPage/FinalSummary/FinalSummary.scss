.final-summary {
  .abstract {
    white-space: pre-wrap;
    word-wrap: break-word
  }

  .success-message {
    text-align: center;
  }
}
