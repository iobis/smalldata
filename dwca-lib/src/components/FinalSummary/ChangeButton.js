import React from 'react'
import { useTranslation } from 'react-i18next'
import PropTypes from 'prop-types'

export default function ChangeButton({ onClick }) {
  const { t } = useTranslation()
  return (
    <button className="change-button button" onClick={onClick}>
      {t('common.change')}
    </button>
  )
}

ChangeButton.propTypes = {
  onClick: PropTypes.func.isRequired
}
