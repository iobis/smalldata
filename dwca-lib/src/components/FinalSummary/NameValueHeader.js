import React from 'react'
import { useTranslation } from 'react-i18next'

export default function NameValueHeader() {
  const { t } = useTranslation()
  return (
    <thead>
      <tr>
        <th className="type">{t('common.name')}</th>
        <th className="value">{t('common.value')}</th>
      </tr>
    </thead>
  )
}
