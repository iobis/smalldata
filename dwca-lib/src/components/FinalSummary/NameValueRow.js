import classNames from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'

export default function NameValueRow({ className, name, value }) {
  return (
    <tr className={classNames('name-value-row fieldrow', className)}>
      <td className="name">{name}</td>
      <td className="value">{!value ? 'â€”' : value}</td>
    </tr>
  )
}

NameValueRow.propTypes = {
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
  name:      PropTypes.string.isRequired,
  value:     PropTypes.oneOfType([PropTypes.string, PropTypes.number])
}
