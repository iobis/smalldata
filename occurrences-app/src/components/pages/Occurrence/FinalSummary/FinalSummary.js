import PropTypes from 'prop-types'
import React from 'react'
import { darwinCoreFieldShape } from '../DarwinCoreFields/DarwinCoreFields'
import { datasetShape } from '../SelectDataset/SelectDataset'
import { format } from 'date-fns'
import { locationDataShape } from '../LocationData/LocationData'
import { observationDataShape } from '../ObservationData/ObservationData'
import { occurrenceDataShape } from '../OccurrenceData/OccurrenceData'
import { useTranslation } from 'react-i18next'

export default function FinalSummary({
  dataset,
  darwinCoreFields,
  occurrenceData,
  observationData,
  locationData,
  measurements,
  onChangeClick,
  onSubmitClick
}) {
  const { t } = useTranslation()

  return (
    <div className="final-summary section is-fluid">
      <div className="columns is-centered">
        <h1 className="final-summary-title title is-3">{t('occurrenceForm.finalSummary.title')}</h1>
      </div>
      <SubmitEntryButton onClick={onSubmitClick}/>
      <section className="select-dataset">
        <SectionTitle>1 - {t('occurrenceForm.selectDataset.step.stepTitle')}</SectionTitle>
        <p>{dataset.description}</p>
        <ChangeButton onClick={() => onChangeClick('selectDataset')}/>
      </section>
      <section className="occurrence-data">
        <SectionTitle>2 - {t('occurrenceForm.occurrenceData.step.stepTitle')}</SectionTitle>
        <table className="table is-striped is-fullwidth is-hoverable">
          <tbody>
            <NameValueRow name="scientific name" value={occurrenceData.scientificName}/>
            <NameValueRow name="beginDate" value={format(occurrenceData.beginDate, 'D MMMM YYYY')}/>
            <NameValueRow name="endDate" value={format(occurrenceData.endDate, 'D MMMM YYYY')}/>
            <NameValueRow name="occurrenceStatus" value={occurrenceData.occurrenceStatus}/>
            <NameValueRow name="basisOfRecord" value={occurrenceData.basisOfRecord}/>
            <NameValueRow name="sex" value={occurrenceData.sex}/>
            <NameValueRow name="lifestage" value={occurrenceData.lifestage}/>
          </tbody>
        </table>
        <ChangeButton onClick={() => onChangeClick('occurrenceData')}/>
      </section>
      <section className="location-data">
        <SectionTitle>3 - {t('occurrenceForm.locationData.step.stepTitle')}</SectionTitle>
        <table className="table is-striped is-fullwidth is-hoverable">
          <tbody>
            <NameValueRow name="Latitude" value={locationData.decimalLatitude}/>
            <NameValueRow name="Longitude" value={locationData.decimalLongitude}/>
            <NameValueRow name="coordinateUncertainty" value={locationData.coordinateUncertainty}/>
            <NameValueRow name="minimumDepth" value={locationData.minimumDepth}/>
            <NameValueRow name="maximumDepth" value={locationData.maximumDepth}/>
          </tbody>
        </table>
        <SectionSubtitle>{t('occurrenceForm.finalSummary.locationData.verbatimDataSubtitle')}</SectionSubtitle>
        <table className="table is-striped is-fullwidth is-hoverable">
          <tbody>
            <NameValueRow name="verbatimCoordinates" value={locationData.verbatimCoordinates}/>
            <NameValueRow name="verbatimDepth" value={locationData.verbatimDepth}/>
          </tbody>
        </table>
        <ChangeButton onClick={() => onChangeClick('locationData')}/>
      </section>
      <section className="observation-data">
        <SectionTitle>4 - {t('occurrenceForm.observationData.step.stepTitle')}</SectionTitle>
        <SectionSubtitle>{t('occurrenceForm.finalSummary.observationData.catalogDataSubtitle')}</SectionSubtitle>
        <div className="content">
          <table className="table is-striped is-fullwidth is-hoverable">
            <tbody>
              <NameValueRow name="institutionCode" value={observationData.institutionCode}/>
              <NameValueRow name="collectionCode" value={observationData.collectionCode}/>
              <NameValueRow name="fieldNumber" value={observationData.fieldNumber}/>
              <NameValueRow name="catalogNumber" value={observationData.catalogNumber}/>
              <NameValueRow name="recordNumber" value={observationData.recordNumber}/>
            </tbody>
          </table>
          <div className="columns">
            <div className="column is-3">
              <p className="title is-5">{t('occurrenceForm.observationData.identifiedBy.label')}</p>
              <ul>
                {observationData.identifiedBy.map((name) => <li key={name}>{name}</li>)}
              </ul>
            </div>
            <div className="column is-3">
              <p className="title is-5">{t('occurrenceForm.observationData.recordedBy.label')}</p>
              <ul>
                {observationData.recordedBy.map((name) => <li key={name}>{name}</li>)}
              </ul>
            </div>
          </div>
        </div>
        <SectionSubtitle>{t('occurrenceForm.finalSummary.observationData.speciesDataSubtitle')}</SectionSubtitle>
        <table className="table is-striped is-fullwidth is-hoverable">
          <tbody>
            <NameValueRow name="references" value={observationData.references.join(', ')}/>
            <NameValueRow name="identificationQualifier" value={observationData.identificationQualifier}/>
            <NameValueRow name="identificationRemarks" value={observationData.identificationRemarks}/>
          </tbody>
        </table>
        <ChangeButton onClick={() => onChangeClick('observationData')}/>
      </section>
      <section className="measurement-or-fact">
        <SectionTitle>5 - {t('occurrenceForm.measurementOrFact.step.stepTitle')}</SectionTitle>
        <table className="measurements table is-fullwidth is-striped is-hoverable">
          <thead>
            <tr>
              <th className="type">{t('common.type')}</th>
              <th className="unit">{t('common.unit')}</th>
              <th className="value">{t('common.value')}</th>
            </tr>
          </thead>
          <tbody>
            {measurements.map(({ type, unit, value }, index) => (
              // eslint-disable-next-line react/no-array-index-key
              <tr className="fieldrow" key={index}>
                <td>{type}</td>
                <td>{unit}</td>
                <td>{value}</td>
              </tr>
            ))}
          </tbody>
        </table>
        <ChangeButton onClick={() => onChangeClick('measurementOrFact')}/>
      </section>
      <section className="darwin-core-fields">
        <SectionTitle>6 - {t('occurrenceForm.darwinCoreFields.step.stepTitle')}</SectionTitle>
        <table className="table is-striped is-fullwidth is-hoverable">
          <tbody>
            {darwinCoreFields.map((field, index) =>
              // eslint-disable-next-line react/no-array-index-key
              <NameValueRow key={index} {...field}/>
            )}
          </tbody>
        </table>
        <ChangeButton onClick={() => onChangeClick('darwinCoreFields')}/>
      </section>
      <SubmitEntryButton onClick={onSubmitClick}/>
    </div>
  )
}

FinalSummary.propTypes = {
  darwinCoreFields: PropTypes.arrayOf(PropTypes.shape(darwinCoreFieldShape)).isRequired,
  dataset:          PropTypes.shape(datasetShape).isRequired,
  locationData:     PropTypes.shape(locationDataShape).isRequired,
  measurements:     PropTypes.array.isRequired,
  observationData:  PropTypes.shape(observationDataShape).isRequired,
  occurrenceData:   PropTypes.shape(occurrenceDataShape).isRequired,
  onChangeClick:    PropTypes.func.isRequired,
  onSubmitClick:    PropTypes.func.isRequired
}

function NameValueRow({ name, value }) {
  return (
    <tr className="name-value-row fieldrow">
      <td className="name">{name}</td>
      <td className="value">{value}</td>
    </tr>
  )
}

NameValueRow.propTypes = {
  name:  PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
}

function SubmitEntryButton({ onClick }) {
  const { t } = useTranslation()
  return (
    <div className="submit-entry-button columns is-centered">
      <button className="button is-info is-medium" onClick={onClick}>
        {t('occurrenceForm.finalSummary.submitEntryButton')}
      </button>
    </div>
  )
}

SubmitEntryButton.propTypes = {
  onClick: PropTypes.func.isRequired
}

function SectionTitle({ children }) {
  return (
    <h2 className="title is-4">{children}</h2>
  )
}

SectionTitle.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired
}

function ChangeButton({ onClick }) {
  const { t } = useTranslation()
  return (
    <button className="change-button button" onClick={onClick}>
      {t('common.change')}
    </button>
  )
}

ChangeButton.propTypes = {
  onClick: PropTypes.func.isRequired
}

function SectionSubtitle({ children }) {
  return <h2 className="title is-5">{children}</h2>
}

SectionSubtitle.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired
}
