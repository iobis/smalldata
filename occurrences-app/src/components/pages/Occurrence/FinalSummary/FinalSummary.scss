.final-summary {
  .final-summary-title {
    margin-bottom: 1.5rem;
  }

  .submit-entry-button {
    margin-bottom: 1.5rem;
  }

  section {
    margin-bottom: 1.5rem;

    &.dataset-summary {
      .change-button {
        margin-top: 1rem;
      }
    }
  }

  .name-value-row {
    .name {
      width: 30%;
    }

    .value {
      width: 70%;
    }
  }

  .measurements {
    .type {
      width: 30%
    }

    .unit {
      width: 20%
    }

    .value {
      width: 20%
    }
  }

  .success-message {
    text-align: center;

    .create-fresh {
      margin-right: 1rem;
    }
  }
}
