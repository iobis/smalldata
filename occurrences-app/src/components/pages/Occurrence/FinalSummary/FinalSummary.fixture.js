import * as SmalldataClient from '../../../../clients/SmalldataClient'
import * as SmalldataClientFixture from '../../../../clients/SmalldataClient.mock'
import FinalSummary from './FinalSummary'

export default [{
  component: FinalSummary,
  name:      'default',
  url:       '/input-data/new',
  props:     {
    ...getDefaultProps()
  }
}, {
  component: FinalSummary,
  name:      'with error',
  url:       '/input-data/new',
  props:     {
    ...getDefaultProps(),
    errorVisible: true,
    errorMessage: 'error message'
  }
}, {
  component: FinalSummary,
  name:      'successfully submitted',
  url:       '/input-data/new',
  props:     {
    ...getDefaultProps(),
    successVisible: true
  }
}]

export function getDefaultProps() {
  return {
    dataset:          SmalldataClientFixture.DATASTES_RESPONSE.map(SmalldataClient.renameRefToId)[0],
    occurrenceData:   {
      basisOfRecord:    'humanObservation',
      beginDate:        Date.UTC(2019, 3, 29),
      endDate:          Date.UTC(2019, 3, 30),
      lifestage:        'adult',
      occurrenceStatus: 'absent',
      scientificName:   'ala abra',
      sex:              'male'
    },
    locationData:     {
      decimalLongitude:      2.345456,
      decimalLatitude:       51.3354656,
      coordinateUncertainty: null,
      minimumDepth:          null,
      maximumDepth:          null,
      verbatimCoordinates:   '41 05 54S 121 05 34W',
      verbatimDepth:         '100 - 200 m'
    },
    observationData:  {
      institutionCode:         'IBSS',
      collectionCode:          'R/V N. Danilevskiy 1935 Azov Sea benthos data',
      fieldNumber:             '557',
      catalogNumber:           'IBSS_Benthos_1935_1331',
      recordNumber:            '123456',
      identifiedBy:            ['Indiana Jones'],
      recordedBy:              ['Harrison Ford'],
      identificationQualifier: 'some identification qualifier',
      identificationRemarks:   'some identification remarks',
      references:              ['http://www.google.com', 'https://clojure.org/']
    },
    measurements:     [
      { type: 'Pressure', unit: 'Decibars', value: '10' },
      { type: 'Pressure', unit: 'Decibars', value: '50' }
    ],
    darwinCoreFields: [
      { name: 'name-1', value: 'value-1' },
      { name: 'name-2', value: 'value-2' },
      { name: 'name-3', value: 'value-3' }
    ],
    onChangeClick:    console.log,
    onSubmitClick:    console.log,
    onErrorClose:     console.log,
    errorVisible:     false
  }
}
