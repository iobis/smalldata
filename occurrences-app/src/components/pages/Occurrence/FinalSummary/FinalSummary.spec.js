import FinalSummary from './FinalSummary'
import React from 'react'
import { getDefaultProps } from './FinalSummary.fixture'
import { mount } from 'enzyme'

describe('FinalSummary', () => {
  it('renders correctly', () => {
    expect(mount(createComponent())).toMatchSnapshot()
  })

  describe('when clicking change button', () => {
    [{ className: 'select-dataset', event: 'selectDataset' },
      { className: 'occurrence-data', event: 'occurrenceData' },
      { className: 'location-data', event: 'locationData' },
      { className: 'observation-data', event: 'observationData' },
      { className: 'measurement-or-fact', event: 'measurementOrFact' },
      { className: 'darwin-core-fields', event: 'darwinCoreFields' }
    ].forEach(({ className, event }) => {
      it(`calls onChangeClick handler with ${event} when clicking change button in ${className}`, () => {
        const onChangeClick = jest.fn()
        const wrapper = mount(createComponent({ onChangeClick }))
        wrapper.find('.' + className + ' .change-button').simulate('click')
        expect(onChangeClick).toHaveBeenCalledTimes(1)
        expect(onChangeClick).toBeCalledWith(event)
      })
    })
  })
})

function createComponent(props) {
  const defaultProps = {
    ...getDefaultProps(),
    onChangeClick: jest.fn(),
    onSubmitClick: jest.fn()
  }

  return <FinalSummary {...defaultProps} {...props}/>
}
