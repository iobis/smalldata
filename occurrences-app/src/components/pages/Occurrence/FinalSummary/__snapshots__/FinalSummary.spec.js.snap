// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalSummary renders correctly 1`] = `
<FinalSummary
  darwinCoreFields={
    Array [
      Object {
        "name": "name-1",
        "value": "value-1",
      },
      Object {
        "name": "name-2",
        "value": "value-2",
      },
      Object {
        "name": "name-3",
        "value": "value-3",
      },
    ]
  }
  dataset={
    Object {
      "description": "NPPSD Short-tailed Albatross Sightings",
      "id": 1,
    }
  }
  locationData={
    Object {
      "coordinateUncertainty": null,
      "decimalLatitude": 51.3354656,
      "decimalLongitude": 2.345456,
      "maximumDepth": null,
      "minimumDepth": null,
      "verbatimCoordinates": "41 05 54S 121 05 34W",
      "verbatimDepth": "100 - 200 m",
    }
  }
  measurements={
    Array [
      Object {
        "type": "Pressure",
        "unit": "Decibars",
        "value": "10",
      },
      Object {
        "type": "Pressure",
        "unit": "Decibars",
        "value": "50",
      },
    ]
  }
  observationData={
    Object {
      "catalogNumber": "IBSS_Benthos_1935_1331",
      "collectionCode": "R/V N. Danilevskiy 1935 Azov Sea benthos data",
      "fieldNumber": "557",
      "identificationQualifier": "some identification qualifier",
      "identificationRemarks": "some identification remarks",
      "identifiedBy": Array [
        "Indiana Jones",
      ],
      "institutionCode": "IBSS",
      "recordNumber": "123456",
      "recordedBy": Array [
        "Harrison Ford",
      ],
      "references": Array [
        "http://www.google.com",
        "https://clojure.org/",
      ],
    }
  }
  occurrenceData={
    Object {
      "basisOfRecord": "humanObservation",
      "beginDate": 1556496000000,
      "endDate": 1556582400000,
      "lifestage": "adult",
      "occurrenceStatus": "absent",
      "scientificName": "ala abra",
      "sex": "male",
    }
  }
  onChangeClick={[MockFunction]}
  onSubmitClick={[MockFunction]}
>
  <div
    className="final-summary section is-fluid"
  >
    <div
      className="columns is-centered"
    >
      <h1
        className="final-summary-title title is-3"
      >
        occurrenceForm.finalSummary.title
      </h1>
    </div>
    <SubmitEntryButton
      onClick={[MockFunction]}
    >
      <div
        className="submit-entry-button columns is-centered"
      >
        <button
          className="button is-info is-medium"
          onClick={[MockFunction]}
        >
          occurrenceForm.finalSummary.submitEntryButton
        </button>
      </div>
    </SubmitEntryButton>
    <section
      className="select-dataset"
    >
      <SectionTitle>
        <h2
          className="title is-4"
        >
          1 - 
          occurrenceForm.selectDataset.step.stepTitle
        </h2>
      </SectionTitle>
      <p>
        NPPSD Short-tailed Albatross Sightings
      </p>
      <ChangeButton
        onClick={[Function]}
      >
        <button
          className="change-button button"
          onClick={[Function]}
        >
          common.change
        </button>
      </ChangeButton>
    </section>
    <section
      className="occurrence-data"
    >
      <SectionTitle>
        <h2
          className="title is-4"
        >
          2 - 
          occurrenceForm.occurrenceData.step.stepTitle
        </h2>
      </SectionTitle>
      <table
        className="table is-striped is-fullwidth is-hoverable"
      >
        <tbody>
          <NameValueRow
            name="scientific name"
            value="ala abra"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                scientific name
              </td>
              <td
                className="value"
              >
                ala abra
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="beginDate"
            value="29 April 2019"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                beginDate
              </td>
              <td
                className="value"
              >
                29 April 2019
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="endDate"
            value="30 April 2019"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                endDate
              </td>
              <td
                className="value"
              >
                30 April 2019
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="occurrenceStatus"
            value="absent"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                occurrenceStatus
              </td>
              <td
                className="value"
              >
                absent
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="basisOfRecord"
            value="humanObservation"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                basisOfRecord
              </td>
              <td
                className="value"
              >
                humanObservation
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="sex"
            value="male"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                sex
              </td>
              <td
                className="value"
              >
                male
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="lifestage"
            value="adult"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                lifestage
              </td>
              <td
                className="value"
              >
                adult
              </td>
            </tr>
          </NameValueRow>
        </tbody>
      </table>
      <ChangeButton
        onClick={[Function]}
      >
        <button
          className="change-button button"
          onClick={[Function]}
        >
          common.change
        </button>
      </ChangeButton>
    </section>
    <section
      className="location-data"
    >
      <SectionTitle>
        <h2
          className="title is-4"
        >
          3 - 
          occurrenceForm.locationData.step.stepTitle
        </h2>
      </SectionTitle>
      <table
        className="table is-striped is-fullwidth is-hoverable"
      >
        <tbody>
          <NameValueRow
            name="Latitude"
            value={51.3354656}
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                Latitude
              </td>
              <td
                className="value"
              >
                51.3354656
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="Longitude"
            value={2.345456}
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                Longitude
              </td>
              <td
                className="value"
              >
                2.345456
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="coordinateUncertainty"
            value={null}
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                coordinateUncertainty
              </td>
              <td
                className="value"
              />
            </tr>
          </NameValueRow>
          <NameValueRow
            name="minimumDepth"
            value={null}
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                minimumDepth
              </td>
              <td
                className="value"
              />
            </tr>
          </NameValueRow>
          <NameValueRow
            name="maximumDepth"
            value={null}
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                maximumDepth
              </td>
              <td
                className="value"
              />
            </tr>
          </NameValueRow>
        </tbody>
      </table>
      <SectionSubtitle>
        <h2
          className="title is-5"
        >
          occurrenceForm.finalSummary.locationData.verbatimDataSubtitle
        </h2>
      </SectionSubtitle>
      <table
        className="table is-striped is-fullwidth is-hoverable"
      >
        <tbody>
          <NameValueRow
            name="verbatimCoordinates"
            value="41 05 54S 121 05 34W"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                verbatimCoordinates
              </td>
              <td
                className="value"
              >
                41 05 54S 121 05 34W
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="verbatimDepth"
            value="100 - 200 m"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                verbatimDepth
              </td>
              <td
                className="value"
              >
                100 - 200 m
              </td>
            </tr>
          </NameValueRow>
        </tbody>
      </table>
      <ChangeButton
        onClick={[Function]}
      >
        <button
          className="change-button button"
          onClick={[Function]}
        >
          common.change
        </button>
      </ChangeButton>
    </section>
    <section
      className="observation-data"
    >
      <SectionTitle>
        <h2
          className="title is-4"
        >
          4 - 
          occurrenceForm.observationData.step.stepTitle
        </h2>
      </SectionTitle>
      <SectionSubtitle>
        <h2
          className="title is-5"
        >
          occurrenceForm.finalSummary.observationData.catalogDataSubtitle
        </h2>
      </SectionSubtitle>
      <div
        className="content"
      >
        <table
          className="table is-striped is-fullwidth is-hoverable"
        >
          <tbody>
            <NameValueRow
              name="institutionCode"
              value="IBSS"
            >
              <tr
                className="name-value-row fieldrow"
              >
                <td
                  className="name"
                >
                  institutionCode
                </td>
                <td
                  className="value"
                >
                  IBSS
                </td>
              </tr>
            </NameValueRow>
            <NameValueRow
              name="collectionCode"
              value="R/V N. Danilevskiy 1935 Azov Sea benthos data"
            >
              <tr
                className="name-value-row fieldrow"
              >
                <td
                  className="name"
                >
                  collectionCode
                </td>
                <td
                  className="value"
                >
                  R/V N. Danilevskiy 1935 Azov Sea benthos data
                </td>
              </tr>
            </NameValueRow>
            <NameValueRow
              name="fieldNumber"
              value="557"
            >
              <tr
                className="name-value-row fieldrow"
              >
                <td
                  className="name"
                >
                  fieldNumber
                </td>
                <td
                  className="value"
                >
                  557
                </td>
              </tr>
            </NameValueRow>
            <NameValueRow
              name="catalogNumber"
              value="IBSS_Benthos_1935_1331"
            >
              <tr
                className="name-value-row fieldrow"
              >
                <td
                  className="name"
                >
                  catalogNumber
                </td>
                <td
                  className="value"
                >
                  IBSS_Benthos_1935_1331
                </td>
              </tr>
            </NameValueRow>
            <NameValueRow
              name="recordNumber"
              value="123456"
            >
              <tr
                className="name-value-row fieldrow"
              >
                <td
                  className="name"
                >
                  recordNumber
                </td>
                <td
                  className="value"
                >
                  123456
                </td>
              </tr>
            </NameValueRow>
          </tbody>
        </table>
        <div
          className="columns"
        >
          <div
            className="column is-3"
          >
            <p
              className="title is-5"
            >
              occurrenceForm.observationData.identifiedBy.label
            </p>
            <ul>
              <li
                key="Indiana Jones"
              >
                Indiana Jones
              </li>
            </ul>
          </div>
          <div
            className="column is-3"
          >
            <p
              className="title is-5"
            >
              occurrenceForm.observationData.recordedBy.label
            </p>
            <ul>
              <li
                key="Harrison Ford"
              >
                Harrison Ford
              </li>
            </ul>
          </div>
        </div>
      </div>
      <SectionSubtitle>
        <h2
          className="title is-5"
        >
          occurrenceForm.finalSummary.observationData.speciesDataSubtitle
        </h2>
      </SectionSubtitle>
      <table
        className="table is-striped is-fullwidth is-hoverable"
      >
        <tbody>
          <NameValueRow
            name="references"
            value="http://www.google.com, https://clojure.org/"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                references
              </td>
              <td
                className="value"
              >
                http://www.google.com, https://clojure.org/
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="identificationQualifier"
            value="some identification qualifier"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                identificationQualifier
              </td>
              <td
                className="value"
              >
                some identification qualifier
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            name="identificationRemarks"
            value="some identification remarks"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                identificationRemarks
              </td>
              <td
                className="value"
              >
                some identification remarks
              </td>
            </tr>
          </NameValueRow>
        </tbody>
      </table>
      <ChangeButton
        onClick={[Function]}
      >
        <button
          className="change-button button"
          onClick={[Function]}
        >
          common.change
        </button>
      </ChangeButton>
    </section>
    <section
      className="measurement-or-fact"
    >
      <SectionTitle>
        <h2
          className="title is-4"
        >
          5 - 
          occurrenceForm.measurementOrFact.step.stepTitle
        </h2>
      </SectionTitle>
      <table
        className="measurements table is-fullwidth is-striped is-hoverable"
      >
        <thead>
          <tr>
            <th
              className="type"
            >
              common.type
            </th>
            <th
              className="unit"
            >
              common.unit
            </th>
            <th
              className="value"
            >
              common.value
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="fieldrow"
            key="0"
          >
            <td>
              Pressure
            </td>
            <td>
              Decibars
            </td>
            <td>
              10
            </td>
          </tr>
          <tr
            className="fieldrow"
            key="1"
          >
            <td>
              Pressure
            </td>
            <td>
              Decibars
            </td>
            <td>
              50
            </td>
          </tr>
        </tbody>
      </table>
      <ChangeButton
        onClick={[Function]}
      >
        <button
          className="change-button button"
          onClick={[Function]}
        >
          common.change
        </button>
      </ChangeButton>
    </section>
    <section
      className="darwin-core-fields"
    >
      <SectionTitle>
        <h2
          className="title is-4"
        >
          6 - 
          occurrenceForm.darwinCoreFields.step.stepTitle
        </h2>
      </SectionTitle>
      <table
        className="table is-striped is-fullwidth is-hoverable"
      >
        <tbody>
          <NameValueRow
            key="0"
            name="name-1"
            value="value-1"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                name-1
              </td>
              <td
                className="value"
              >
                value-1
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            key="1"
            name="name-2"
            value="value-2"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                name-2
              </td>
              <td
                className="value"
              >
                value-2
              </td>
            </tr>
          </NameValueRow>
          <NameValueRow
            key="2"
            name="name-3"
            value="value-3"
          >
            <tr
              className="name-value-row fieldrow"
            >
              <td
                className="name"
              >
                name-3
              </td>
              <td
                className="value"
              >
                value-3
              </td>
            </tr>
          </NameValueRow>
        </tbody>
      </table>
      <ChangeButton
        onClick={[Function]}
      >
        <button
          className="change-button button"
          onClick={[Function]}
        >
          common.change
        </button>
      </ChangeButton>
    </section>
    <SubmitEntryButton
      onClick={[MockFunction]}
    >
      <div
        className="submit-entry-button columns is-centered"
      >
        <button
          className="button is-info is-medium"
          onClick={[MockFunction]}
        >
          occurrenceForm.finalSummary.submitEntryButton
        </button>
      </div>
    </SubmitEntryButton>
  </div>
</FinalSummary>
`;
